creator-node/src/routes/files.js

```diff
@@ -247,46 +254,97 @@ module.exports = function (app) {
     // Do not act as a public gateway. Only serve IPFS files that are tracked by this creator node.
     const dirCID = req.params.dirCID
     const filename = req.params.filename
+    const ipfsPath = `${dirCID}/${filename}`
 
     const queryResults = await models.File.findOne({ where: {
-      multihash: dirCID,
-      type: 'dir'
+      dirMultihash: dirCID,
+      // All other file types are valid and can be served through this route.
```

Comment:
is this true? this route was originally intended only for multi res images since other tyeps (tracks, metadata) don't have filenames. i think this might work since filename would be = sourceFile for all other types now, but not sure if this is what you intend. you'd have to pass dirCID as empty string / null to get that to work. also below code to parse parentStoragePath would break for non-dir images p sure

Reply:
not sure if i understand this comment. this is the *dir* CID path, but the query is querying for the "leaf" file. Copied comment is weird though, you're right. Cleaning that up & adding a new comment.